v1/messages/src/index.ts

```diff
@@ -6,31 +6,33 @@ import {
   handlePutRequest,
   parseSlugFromURI,
   MessagePayload,
+  parseAuthorizationHeader,
+  isUserAuth,
 } from "./requestHandlers";
 
 export const handleRequest: HandleRequest = async function (
   request: HttpRequest
 ): Promise<HttpResponse> {
   const method = request.method;
-  const uri = request.uri;
-  const msgID = parseSlugFromURI(uri, "messages");
+  const url = request.headers["spin-full-url"];
+  const msgID = parseSlugFromURI(url, "messages");
+  const auth = request.headers.authorization ?? null;
+  const authID = parseAuthorizationHeader(auth);
 
-  if (method == "GET") {
+  if (authID === null) {
+    return { status: 400 };
```

Comment:
If you have a null authID, that counts as being unauthorized. while 400 is appropriate as well, but it is generally used to represent a generic user error, when a more specific code doesn't exist for it. If we know the error is specific to user authorization, 401 is better.

Both `400 BAD REQUEST` and `500 INTERNAL SERVER` function this way. They are the first error for their range, and represent generic errors.

Reply:
PS: If you implement my suggestion [here](https://github.com/mabanana/crud-list/pull/2#discussion_r1628074839) then you wouldn't need this if statement altogether, because all of the header parsing logic would be encapsulated inside `isUserAuth`